integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/ExcludeHeadersWithCacheTestCase.java

```diff
@@ -0,0 +1,87 @@
+ * under the License.
+ */
+package org.wso2.carbon.esb.mediator.test.cache;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.AxisFault;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+
+import javax.xml.namespace.QName;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Cache Mediator - verify 'headersToExcludeInHash' parameter
+ *
+ * This testcase is to verify that a defined header can be exclude when hashing
+ */
+public class ExcludeHeadersWithCacheTestCase extends ESBIntegrationTest{
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+    }
+
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    @Test(groups = "wso2.esb", description = "Verify a header can be excluded when hashing")
+    public void testExcludingHeaders() throws AxisFault {
+        OMElement stockResponse;
+
+        // Sending a request to SimpleStockQuote Service
+        stockResponse = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("excludeHeaders"), "", "ExcludeHeaders");
+        assertNotNull(stockResponse, "Response is null");
+        String change = stockResponse.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        // Sending a request to SimpleStockQuote Service with 'user' header
+        axis2Client.addHttpHeader("user", "Peter");
+        stockResponse = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("excludeHeaders"), "", "ExcludeHeaders");
+        assertNotNull(stockResponse, "Response is null");
+        String change1 = stockResponse.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        assertTrue(change.equals(change1), "Response caching did not exclude the given header value");
+
+        // Sending a request to SimpleStockQuote Service with 'user' and 'country' headers
+        axis2Client.addHttpHeader("user", "Peter");
+        axis2Client.addHttpHeader("country","UK");
+        stockResponse = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("excludeHeaders"), "", "ExcludeHeaders");
+        assertNotNull(stockResponse, "Response is null");
+        String change2 = stockResponse.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        assertTrue(!change.equals(change2), "Response caching works with the undefined exclude header");
+
+        log.info("In Initial Message :" + change);
```

Comment:
Do we need these logs after the assertion has passed?

Reply:
Will remove those